MobileApplication/MobileApplication/ViewModels/Goal/GoalDetailViewModel.cs

```diff
@@ -118,6 +136,91 @@ private async void OnDelete()
 
             IsBusy = false;
         }
+
+        public string GoalPossibility(decimal worth)
+        {
+            decimal profitAWeek = profit / 4;
+            decimal possibilityRate = worth / profitAWeek;
+            Debug.WriteLine(possibilityRate);
+
+            if (possibilityRate < (decimal)0.5)
+            {
+                return "Huge";
+            }
+
+            if (possibilityRate < (decimal)0.8)
+            {
+                return "Real";
+            }
+
+            if (possibilityRate < 1)
+            {
+                return "Small";
+            }
+
+            return "Not real";
+        }
+
+
+        public void CalculatePossibility()
+        {
+            decimal money;
+            profit = 0;
+            if (DeadlineDate.Subtract(DateTime.UtcNow).Days > 7)
+            {
+                money = Decimal.Parse(Amount) / (DeadlineDate.Subtract(DateTime.UtcNow).Days / 7);
+            }
+            else
+            {
+                money = Decimal.Parse(Amount);
+            }
+
+            CalculateProfit();
+            Possibility = GoalPossibility(money);
+        }
+
+        public void CalculateProfit()
+        {
+            try
+            {
+                List<string> months = new List<string>();
+                foreach (Transaction transaction in Transactions)
+                {
+                    string date = transaction.TrTime.ToString("yyyyMM");
+                    if (!months.Contains(date))
+                    {
+                        months.Add(date);
+                    }
+                    profit += transaction.Amount;
+                    profit /= months.Count;
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex);
+            }
+        }
+
+        async Task ExecuteLoadTransactionItemsCommand()
+        {
+            IsBusy = true;
+            Transactions.Clear();
+
+            try
+            {
+                var items = await TransactionRestService.RefreshDataAsync();
+                items.ForEach(transaction => Transactions.Add(transaction));
```

Comment:
Couldn't it be simplified to `Transactions.AddRange(items)`?

Reply:
ObservableCollection has no AddRange method. I don't think and extension method is necessary if I'd only use it once.